src/genericactiondepotlib/CMakeLists.txt

```diff
@@ -18,14 +31,18 @@ include_directories(${Boost_INCLUDE_DIRS})
 add_library(${PROJECT_NAME} MODULE ${allSrc})
 set_target_properties(${PROJECT_NAME} PROPERTIES PREFIX "")
 
-target_include_directories(${PROJECT_NAME} PRIVATE ${PROJECT_SOURCE_DIR}/src ${PROJECT_SOURCE_DIR}/../src)
-
-target_link_libraries(${PROJECT_NAME} blocktestcore ${Boost_LIBRARIES})
+target_include_directories(${PROJECT_NAME} PRIVATE ${PROJECT_SOURCE_DIR}/src ${PROJECT_SOURCE_DIR}/../src ${serial_SOURCE_DIR}/include )
```

Comment:
I don't think `${serial_SOURCE_DIR}/include` is necessary here.

Reply:
Also, now that I notice that, if it is actually required, it should be in the `if(USE_SERIAL_LIB)` below

Reply:
I think that @traversaro  has correct on the fly the serial library, because now without ${serial_SOURCE_DIR}/include all is working. :)
Also, @drdanz was a correct observation but now is obsolete.
thanks!